pallets/nfts/src/tests.rs

```diff
@@ -2604,136 +2606,103 @@ fn approve_collection_transfer_works() {
-				collection_id,
-				delegate.clone(),
-				None
-			),
-			Error::<Test>::NoItemOwned
-		);
-
-		// Force-approve collection without items, throws error `Error::NoItemOwned`.
-		assert_noop!(
-			Nfts::force_approve_collection_transfer(
-				RuntimeOrigin::root(),
-				item_owner.clone(),
-				collection_id,
-				delegate.clone(),
-				None
-			),
-			Error::<Test>::NoItemOwned
-		);
-
-		assert_ok!(Nfts::force_mint(
-			RuntimeOrigin::signed(collection_owner.clone()),
-			collection_id,
-			item_id,
-			item_owner.clone(),
-			default_item_config()
-		));
-		assert_ok!(Nfts::force_mint(
-			RuntimeOrigin::signed(collection_owner.clone()),
-			locked_collection_id,
-			item_id,
-			item_owner.clone(),
-			default_item_config()
-		));
-
 		for (origin, maybe_item_owner) in [
 			// Parameters for `approve_collection_transfer`.
 			(RuntimeOrigin::signed(item_owner.clone()), None),
 			// Parameters for `force_approve_collection_transfer`.
-			(root(), Some(item_owner.clone())),
+			(root(), Some(&item_owner)),
 		] {
-			// Throws error `Error::ItemsNonTransferable`.
-			assert_ok!(Nfts::lock_collection(
-				RuntimeOrigin::signed(collection_owner.clone()),
-				locked_collection_id,
-				CollectionSettings::from_disabled(CollectionSetting::TransferableItems.into())
-			));
+			// Approve unknown collection, throws error `Error::NoItemOwned`.
 			assert_noop!(
 				approve_collection_transfer(
 					origin.clone(),
-					maybe_item_owner.clone(),
-					locked_collection_id,
-					delegate.clone(),
+					maybe_item_owner,
+					collection_id,
+					&delegate,
 					None
 				),
-				Error::<Test>::ItemsNonTransferable
+				Error::<Test>::NoItemOwned
 			);
+			assert_ok!(Nfts::force_create(
+				RuntimeOrigin::root(),
+				collection_owner.clone(),
+				default_collection_config()
+			));
 
-			assert_ok!(approve_collection_transfer(
-				origin.clone(),
-				maybe_item_owner.clone(),
+			// Approve collection without items, throws error `Error::NoItemOwned`.
+			assert_noop!(
+				Nfts::approve_collection_transfer(
+					RuntimeOrigin::signed(item_owner.clone()),
+					collection_id,
+					delegate.clone(),
+					None
+				),
+				Error::<Test>::NoItemOwned
+			);
+			assert_ok!(Nfts::force_mint(
+				RuntimeOrigin::signed(collection_owner.clone()),
 				collection_id,
-				delegate.clone(),
-				None
+				item_id,
+				item_owner.clone(),
+				default_item_config()
 			));
-			assert!(events().contains(&Event::<Test>::TransferApproved {
-				collection: collection_id,
-				item: None,
-				owner: item_owner.clone(),
-				delegate: delegate.clone(),
-				deadline: None,
-			}));
-			assert_eq!(Balances::reserved_balance(&item_owner), 1);
 
-			// Approve same delegate again not updating the total reserved funds.
-			assert_ok!(approve_collection_transfer(
-				origin.clone(),
-				maybe_item_owner.clone(),
+			for deadline in [None, None, Some(deadline), Some(deadline * 2), Some(deadline), None] {
```

Comment:
Sorry but I don't understand what this means.

Reply:
see latest commit

Reply:
This is sadly still not easy to understand.  Can we use `System::set_block_number(0)` to reset the time and update the list of deadlines to fit with that?

Reply:
See latest commit